python.d/python_modules/third_party/boinc_client.py

```diff
@@ -0,0 +1,762 @@
+            GUI_RPC_PASSWD_FILE. If file can't be read (not found or no
+            permission to read), try to authorize with a blank password.
+            If authorization is requested and fails, all subsequent calls
+            will be refused with socket.error 'Connection reset by peer' (104).
+            Since most local calls do no require authorization, do not attempt
+            it if you're not sure about the password.
+        '''
+        if password is None and not self.hostname:
+            password = read_gui_rpc_password() or ""
+        nonce = self.rpc.call('<auth1/>').text
+        authhash = hashlib.md5('{0}{1}'.format(nonce, password).encode()).hexdigest().lower()
+        reply = self.rpc.call('<auth2><nonce_hash>{0}</nonce_hash></auth2>'.format(authhash))
+
+        if reply.tag == 'authorized':
+            return True
+        else:
+            return False
+
+    def exchange_versions(self):
+        ''' Return VersionInfo instance with core client version info '''
+        return VersionInfo.parse(self.rpc.call('<exchange_versions/>'))
+
+    def get_cc_status(self):
+        ''' Return CcStatus instance containing basic status, such as
+            CPU / GPU / Network active/suspended, etc
+        '''
+        if not self.connected: self.connect()
+        try:
+            return CcStatus.parse(self.rpc.call('<get_cc_status/>'))
+        except socket.error:
+            self.connected = False
+
+    def get_host_info(self):
+        ''' Get information about host hardware and usage. '''
+        return HostInfo.parse(self.rpc.call('<get_host_info/>'))
+
+    def get_tasks(self):
+        ''' Same as get_results(active_only=False) '''
+        return self.get_results(False)
+
+    def get_results(self, active_only=False):
+        ''' Get a list of results.
+            Those that are in progress will have information such as CPU time
+            and fraction done. Each result includes a name;
+            Use CC_STATE::lookup_result() to find this result in the current static state;
+            if it's not there, call get_state() again.
+        '''
+        reply = self.rpc.call("<get_results><active_only>{0}</active_only></get_results>".format(1 if active_only else 0))
+        if not reply.tag == 'results':
+            return []
+
+        results = []
+        for item in list(reply):
+            results.append(Result.parse(item))
+
+        return results
+
+    def set_mode(self, component, mode, duration=0):
+        ''' Do the real work of set_{run,gpu,network}_mode()
+            This method is not part of the original API.
+            Valid components are 'run' (or 'cpu'), 'gpu', 'network' (or 'net')
+        '''
+        component = component.replace('cpu', 'run')
+        component = component.replace('net', 'network')
+        try:
+            reply = self.rpc.call("<set_{0}_mode><{1}/>\n<duration>{2]</duration>\n</set_{0}_mode>".format(component, RunMode.name(mode).lower(), duration))
```

Comment:
`{2]`
typo

Reply:
This is functionally dead code anyway, so I'm planning on removing it.